actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -67,6 +67,8 @@ def formats
 
           v = if params_readable
             Array(Mime[parameters[:format]])
+          elsif format_from_path_extension
+            [Mime[format_from_path_extension]]
```

Comment:
You're calling this method twice without memoizing the result. I'd store the result in a local var or try to memoize in some way.